File: decorator/en_us/ruby/regular.rb
require './polygon_decorator'

# Polygons regular have some especial properties
class Regular < PolygonDecorator

  def valid
    first_side = @sides[0]
    @sides.each do |side|
      if side != first_side
        return false
      end
    end
    return true
  end

end

File: decorator/en_us/python/regular.py
from polygon_decorator import PolygonDecorator
"""
    An regular Polygon have the restriction of that all
    sides have same lenght
"""
class Regular(PolygonDecorator):

    def valid(self):
        first_side = self.sides[0]
        for side in self.sides:
            if side is not first_side:
                return False
            else:
                pass
        # If all sides are equals
        return True
